src/order/category/Preorder.lean

```diff
@@ -0,0 +1,75 @@
+/-
+Copyleft 2020 Johan Commelin. No rights reserved.
+Authors: Johan Commelin
+-/
+
+import category_theory.concrete_category
+import algebra.punit_instances
+
+/-! # Category of preorders -/
+
+open category_theory
+
+/-- The category of preorders. -/
+def Preorder := bundled preorder
+
+/-- Bundled monotone (aka, increasing) function -/
+structure preorder_hom (α β : Type*) [preorder α] [preorder β] :=
+(to_fun   : α → β)
+(monotone : monotone to_fun)
+
+namespace preorder_hom
+variables {α β γ : Type*} [preorder α] [preorder β] [preorder γ]
+
+instance : has_coe_to_fun (preorder_hom α β) :=
+{ F := λ f, α → β,
+  coe := λ f, f.to_fun }
```

Comment:
if you rewrite to `coe := preorder_hom.to_fun`, you can mark `id` with `@[simps]` and you don't need to write `id_apply` yourself, it will be generated.

Reply:
ooh sweet, done